enterprise/core-edge/src/test/java/org/neo4j/coreedge/scenarios/CoreToCoreCopySnapshotIT.java

```diff
@@ -153,6 +154,71 @@ public void shouldBeAbleToDownloadAfterPruning() throws Exception
         assertEquals( DbRepresentation.of( source ), DbRepresentation.of( newDb ) );
     }
 
+    @Test
+    public void shouldBeAbleToDownloadToRejoinedInstanceAfterPruning() throws Exception
+    {
+        // given
+        File dbDir = dir.directory();
+
+        Map<String,String> coreParams = stringMap();
+        int pruneFrequencyMs = 100;
+        coreParams.put( raft_log_pruning_strategy.name(), "keep_none" );
+        coreParams.put( raft_log_pruning_frequency.name(), pruneFrequencyMs + "ms" );
+
+        try ( Cluster cluster = Cluster
+                .start( dbDir, 3, 0, new TestOnlyDiscoveryServiceFactory(), coreParams, emptyMap(),
+                        StandardV3_0.NAME ) )
+        {
+
+            AtomicBoolean running = new AtomicBoolean( true );
+            Thread thread = new Thread()
+            {
+                @Override
+                public synchronized void run()
+                {
+                    while ( running.get() )
+                    {
+                        try
+                        {
+                            cluster.coreTx( ( db, tx ) -> {
+                                Node node = db.createNode();
+                                node.setProperty( "that's a bam", string( 1024 ) );
+                                tx.success();
+                            } );
+                        }
+                        catch ( Exception e )
+                        {
+                            throw new RuntimeException( e );
+                        }
+                    }
+                }
+            };
+
+            thread.start();
+
+            Thread.sleep( pruneFrequencyMs * 100 );
```

Comment:
Yes, to make sure that we get into the situation where a follower rejoins and needs pruned entries. We should refactor it to check the log files and see when that is rather than just waiting 10 seconds, but I would like to do that in a later PR.